Frontend/src/pages/registrations/components/RegistrationList.jsx

```diff
@@ -1,72 +1,72 @@
+    ...competitionInfo.event_ids.map((event_id) => ({
+      text: event_id,
+      cubingIcon: true,
+    })),
     { text: 'Total' },
   ]
   const footer = useMemo(() => {
-    // We have to use a Map instead of an object to preserve event order
-    const eventCounts = heldEvents.reduce((counts, eventId) => {
-      counts.set(eventId, 0)
-      return counts
-    }, new Map())
-    const { newcomers, totalEvents, countrySet } = registrations.reduce(
-      (info, registration) => {
-        if (registration.user.wca_id === null) {
-          info.newcomers++
-        }
-        info.countrySet.add(registration.user.country.iso2)
-        info.totalEvents += registration.event_ids.length
-        heldEvents.forEach((event_id) => {
-          if (registration.event_ids.includes(event_id)) {
-            eventCounts.set(event_id, eventCounts.get(event_id) + 1)
+    if (registrations) {
+      // We have to use a Map instead of an object to preserve event order
+      const eventCounts = competitionInfo.event_ids.reduce(
+        (counts, eventId) => {
+          counts.set(eventId, 0)
+          return counts
+        },
+        new Map()
+      )
+      const { newcomers, totalEvents, countrySet } = registrations.reduce(
+        (info, registration) => {
+          if (registration.user.wca_id === null) {
+            info.newcomers++
           }
-        })
-        return info
-      },
-      { newcomers: 0, totalEvents: 0, countrySet: new Set() }
-    )
-    return [
-      // Potential grammar issues will be fixed when we introduce I18n
-      `${newcomers} First-timers + ${
-        registrations.length - newcomers
-      } Returners = ${registrations.length} People`,
-      `${countrySet.size} Countries`,
-      ...eventCounts.values(),
-      totalEvents,
-    ]
-  }, [registrations, heldEvents])
-  const registrationList = useMemo(
-    () =>
-      registrations.map((registration) => {
+          info.countrySet.add(registration.user.country.iso2)
+          info.totalEvents += registration.event_ids.length
+          competitionInfo.event_ids.forEach((event_id) => {
+            if (registration.event_ids.includes(event_id)) {
+              eventCounts.set(event_id, eventCounts.get(event_id) + 1)
+            }
+          })
+          return info
+        },
+        { newcomers: 0, totalEvents: 0, countrySet: new Set() }
+      )
+      return [
+        // Potential grammar issues will be fixed when we introduce I18n
+        `${newcomers} First-timers + ${
+          registrations.length - newcomers
+        } Returners = ${registrations.length} People`,
+        `${countrySet.size} Countries`,
+        ...eventCounts.values(),
+        totalEvents,
+      ]
+    }
+    return []
+  }, [registrations, competitionInfo.event_ids])
+  const registrationList = useMemo(() => {
+    if (registrations) {
+      return registrations.map((registration) => {
```

Comment:
Could avoid the explicit `if` and `return`:
```
  useMemo(
    () => registrations?.map((registration) => ...
      ...
    ) ?? [],
    [registrations, competitionInfo.event_ids]
  )
```

Reply:
I don't know about transpiling. I don't think I would call it 'good practice' but I see it a lot. It's fine either way, I personally prefer to keep it shorter and avoid the explicit falsey case when it's not important (the UI using `registrationList` won't show up in the falsey case).

Reply:
ok, I will try to use it everywhere where the falsey case is not important